### DISCUSSION TITLE
0.8.2

### DISCUSSION THREAD

**Discussion by dayman40** [microsoft/WindowsAppSDK#1237] [Announcements]
## Windows App SDK 0.8.2 is now available

Windows App SDK 0.8.2 is a stable servicing release that a couple more critical bugs for 0.8.0. 

### Bugs fixed in Windows App SDK 0.8.2:

- Windows App SDK and WinUI 3 are now supported in Visual Studio 2022 Preview 2 and above.
- For .NET apps, you may receive the following error when passing in an array of enums: `Object contains non-primitive or non-blittable data.`
- Writing using the HandWriting Panel inside a textbox causes a crash
- Icons/images always load at their 100% scale value rather than based on the monitor scale value
- Garbage collection of **EventSource\<T\>** causes subsequent failure to unsubscribe handlers (see [GitHub issue](https://github.com/microsoft/CsWinRT/issues/842) for more details)
- Security fix – see [CVE-2021-34533](https://msrc.microsoft.com/update-guide/en-US/vulnerability/CVE-2021-34533) for more details.
- [SwapChainPanel.CompositionScaleChanged](/windows/winui/api/microsoft.ui.xaml.controls.swapchainpanel.compositionscalechanged) sometimes returning incorrect CompositionScale values after changing display scale

The limitations and known issues for v0.8 also apply to v0.8.2, unless marked otherwise in [our documentation](https://docs.microsoft.com/windows/apps/windows-app-sdk/stable-channel#limitations).

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/WindowsAppSDK/releases/tag/v0.8.2'>0.8.2</a>.</em>

**Comment by ivberg** [comment]
Upon upgrade from 0.8.1 to 0.8.2 I get an error on build that I don't see ref'ed in the release notes.

It's a confusing error message because I already have SDK 5.0.400 (the latest available) and the workaround with (WindowsSdkPackageVersion) doesn't seem to work?

Severity	Code	Description	Project	File	Line	Suppression State
Error		This version of Project Reunion requires WinRT.Runtime.dll version 1.3 or greater. 
    Please update to .NET SDK 5.0.205 or 5.0.302 or later.  
    Or add a temporary Microsoft.Windows.SDK.NET.Ref reference of version 10.0.*.18 or later.
    For example,
        &lt;PropertyGroup>
            &lt;WindowsSdkPackageVersion>10.0.&lt;Target Windows SDK Build Number>.18&lt;/WindowsSdkPackageVersion>
        &lt;/PropertyGroup>
	MyAppUI	C:\Users\user\.nuget\packages\microsoft.projectreunion\0.8.2\build\Microsoft.ProjectReunion.targets	21	

As in this doesn't help:
        &lt;PropertyGroup>
            &lt;WindowsSdkPackageVersion>10.0.18362.18/WindowsSdkPackageVersion>
        &lt;/PropertyGroup>

Severity	Code	Description	Project	File	Line	Suppression State
Error	NETSDK1148	A referenced assembly was compiled using a newer version of Microsoft.Windows.SDK.NET.dll. Please update to a newer .NET SDK in order to reference this assembly.	MyAppUI	C:\Program Files\dotnet\sdk\5.0.400\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Windows.targets	55	

=================

Also changing FrameworkReference  in every combination with WindowsSdkPackageVersion didn't resolve the issue:
  &lt;ItemGroup>
      &lt;FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" RuntimeFrameworkVersion="10.0.18362.16" />
      &lt;FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" TargetingPackVersion="10.0.18362.16" />
  &lt;/ItemGroup>

To this:
  &lt;ItemGroup>
      &lt;FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" RuntimeFrameworkVersion="10.0.18362.18" />
      &lt;FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" TargetingPackVersion="10.0.18362.18" />
  &lt;/ItemGroup>

Makes the compile time error go away but there is a runtime error in it's place:
  An assembly specified in the application dependencies manifest (MyAppUI.deps.json) was not found:
    package: 'runtimepack.Microsoft.Windows.SDK.NET.Ref', version: '10.0.18362.18'
    path: 'Microsoft.Windows.SDK.NET.dll'

**Reply by andrewleader** [reply]
@huichen123 @Scottj1s any ideas here?

**Reply by huichen123** [reply]
unfortunately WindowsSdkPackageVersion is not supported until .net sdk 5.0.302. Does rebuild help?

**Reply by Scottj1s** [reply]
>nuget restore -force?

looks like the deps.json is not correct - it should certainly have recorded the dll as coming from the framework package

**Comment by ivberg** [comment]
+ @andrewleader as well
